### PR TITLE
Feature/bphh 1236/external api admin screens

### PR DESCRIPTION
**Pull request by taw27**
## Description
Feature/bphh 1236/external api admin screens
- Adds external api key index screen with a table of api keys for jurisdictions. Viewable as super admin and admin
- Adds ability to enable or disable api keys for a whole jurisdiction as a super admin
- Add's ability to create, update, expire and revoke api key's as super admin and review manager is api keys are enabled for the jurisdiction

## What type of PR is this? (check all applicable)

- [x ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📦 Chore (Release)
- [ ] ✅ Test

## Related Tickets & Documents
- https://hous-bssb.atlassian.net/browse/BPHH-1236
- https://hous-bssb.atlassian.net/browse/BPHH-1237

## Steps to QA
- First enable api keys for a jurisdiction as a Super admin
    - Navigate to jurisdictions page
    -  Select any jurisdiction and click manage -> External Api keys
    - Should navigate to index screen
    - Enable api key's using the switch top right of screen
   
Screenshot 2024-04-25 at 5 12 25 PM
- To perform CRUD operations for external api key when api keys are enabled for the jurisdiction:
    - Navigate to external api key index page 
        - As a super admin - Navigate to Jurisdictions -> Click Manage -> External API keys
        - As a review manager - Navigate to Configuration Management -> Click edit on "Manage Configuration Management"
    - To Create click "Create new API key " button which should open up a modal
        - Add required details then click "Create"
        - On success should navigate to Manage modal
     - To Manage/Edit External API key, click "Manage" on any of the table keys
         - Should open up the Manage modal
         - Should be able to change everything except token, and revoked at
         - Should be able to revoke if already not revoked.  Should show a confirmation modal
         - And save
 
Screenshot 2024-04-25 at 5 22 20 PM

Screenshot 2024-04-25 at 5 22 26 PM

https://github.com/bcgov/HOUS-permit-portal/assets/32022335/a7480500-b3ae-4390-a26d-f8e3a4d19c40

- to disable api keys for a jurisdiction as a Super admin
    - Navigate to jurisdictions page
    - Select any jurisdiction and click manage -> External Api keys
    - Should navigate to index screen
    - Disable api key's using the switch top right of screen
    - The Create and Manage buttons should become disabled
    - Should be the same for review manager once disabled, but should also show a warning banner
 
Screenshot 2024-04-25 at 5 12 00 PM

https://github.com/bcgov/HOUS-permit-portal/assets/32022335/af43b674-aaac-4cbb-9ad2-926ebcd9767a

### CODE REVIEW

**Comment by bkmaibach** [comment]
please add ACTIVE_RECORD_ENCRYPTION env vars to example

File: app/controllers/api/external_api_keys_controller.rb [hunk 1]
```diff
@@ -2,7 +2,20 @@ class Api::ExternalApiKeysController < Api::ApplicationController
   before_action :set_external_api_key, except: %i[index create]
 
   def index
-    @external_api_key = policy_scope(ExternalApiKey)
+    # Only authoris\zed to query own jurisdiction for review managers
+    if current_user.review_manager? && params[:jurisdiction_id].present? &&
+         current_user.jurisdiction_id != params[:jurisdiction_id]
+      raise Pundit::NotAuthorizedError
```
Comment (bkmaibach) [inline app/controllers/api/external_api_keys_controller.rb:8]: You can delegate this check to the correct policy file
and call it like

authorize :external_api_key, :index?

File: app/frontend/components/domains/external-api-key/external-api-key-modal-sub-route.tsx [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+    revokedAt: externalApiKey?.revokedAt,
+  }
+}
+
+export const ExternalApiKeyModalSubRoute = observer(function ExternalApiKeyModalSubRoute() {
+  const { t } = useTranslation()
+  const { externalApiKeyId } = useParams()
+  const navigate = useNavigate()
+  const { jurisdictionStore } = useMst()
+  const { currentJurisdiction } = jurisdictionStore
```
Comment (bkmaibach) [inline app/frontend/components/domains/external-api-key/external-api-key-modal-sub-route.tsx:56]: a better way to get the currentJurisdiction is with the useJurisdiction hook

Comment (bkmaibach) [reply app/frontend/components/domains/external-api-key/external-api-key-modal-sub-route.tsx:56]: unless there is already a parent component that uses it

**Review by bkmaibach — APPROVED** [review]

File: app/frontend/components/domains/external-api-key/external-api-key-modal-sub-route.tsx (hunk 2 above)
Comment (taw27) [reply app/frontend/components/domains/external-api-key/external-api-key-modal-sub-route.tsx:56]: @bkmaibach parent component uses it. This is a subroute that get's rendered in an `Outlet` . https://reactrouter.com/en/main/components/outlet

File: app/controllers/api/external_api_keys_controller.rb (hunk 1 above)
Comment (taw27) [reply app/controllers/api/external_api_keys_controller.rb:8]: @bkmaibach I need the `params`, it's not available in the index? method in the policy

**Pull request merged**